Native compilation again does't like spaces in project directory in 0.15.0 and 0.16.0 ver

**Describe the bug**
Native compilation doesn't like space in project directory. It is a old bug anf fixed in 0.13.0 release reference ticket no: #1673 . i am facing the samer issue with 0.15.0 and 0.16.0 releases.

**Expected behavior**
Native compilation should allow spaces in the project directory path.

**Actual behavior**
Throwing the below error
12:57:08 Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: /home/jenkins/workspace/Online%20Service%20Name%20CRP/NameService/target/NameService-0.0.1-SNAPSHOT-runner.jar (No such file or directory)
12:57:08 	at io.quarkus.undertow.runtime.graal.URLResourceSubstitution$ResourceInfo.<clinit>(URLResourceSubstitution.java:103)
12:57:08 	... 40 more
12:57:08 Caused by: java.io.FileNotFoundException: /home/jenkins/workspace/Online%20Service%20Name%20CRP/NameService/target/NameService-0.0.1-SNAPSHOT-runner.jar (No such file or directory)



**To Reproduce**
Steps to reproduce the behavior:
1. Create project directory name with space and execute a native compile

--- a/extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/graal/URLResourceSubstitution.java
+++ b/extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/graal/URLResourceSubstitution.java
@@ -1,6 +1,7 @@
 package io.quarkus.undertow.runtime.graal;
 
 import java.io.IOException;
+import java.net.URISyntaxException;
 import java.net.URL;
 import java.net.URLClassLoader;
 import java.util.Collections;
@@ -73,7 +74,7 @@ final class URLResourceSubstitution {
             URLClassLoader ucl = (URLClassLoader) cl;
             for (URL res : ucl.getURLs()) {
                 if (res.getProtocol().equals("file") && res.getPath().endsWith(".jar")) {
-                    try (JarFile file = new JarFile(res.getPath())) {
+                    try (JarFile file = new JarFile(res.toURI().getPath())) {
                         Enumeration<JarEntry> e = file.entries();
                         while (e.hasMoreElements()) {
                             JarEntry entry = e.nextElement();
@@ -83,7 +84,7 @@ final class URLResourceSubstitution {
                             }
                         }
 
-                    } catch (IOException e) {
+                    } catch (IOException | URISyntaxException e) {
                         throw new RuntimeException(e);
                     }
                 }

